java-formats/kml-format_kml/src/com/dataiku/dss/utils/KMLParser.java

```diff
@@ -0,0 +1,140 @@
+        logger.info("Children of" + parent);
+        for (int i=0; i < parent.getChildNodes().getLength(); i++){
+            logger.info("Child "+i+":"+parent.getChildNodes().item(i));
+        }
+        NodeList nl = parent.getElementsByTagName(name);
+        if (nl.getLength()==0) return null;
+        else return (Element)nl.item(0);
+    }
+
+    public String getTextContent(Node e){return e.getTextContent();}
+
+    private void putAttrValueIfExists(ColumnFactory cf, Row r, String columnName, Element e, String attrName) {
+        String attrValue = e.getAttribute(attrName);
+        if (!StringUtils.isBlank(attrValue)) {
+            r.put(cf.column(columnName), attrValue);
+        }
+    }
+
+    private void putContentIfExistsInChild(ColumnFactory cf, Row r, String columnName, Element e, String childNodeName) {
+        Node childNode = getFirstNodeByTagName(e, childNodeName);
+        if (childNode != null) {
+            String txt = getTextContent(childNode);
+            if (txt != null) r.put(cf.column(columnName), txt);
+        }
+    }
+
+    private void parsePlacemark(Node node, ProcessorOutput out, ColumnFactory cf, RowFactory rf) throws Exception {
+        Element e = (Element)node;
+        //assert(node.getLocalName().equals("Placemark"));
+
+        Row r = rf.row();
+
+        putContentIfExistsInChild(cf, r, "name", e, "name");
+        cf.column("id");
+        putAttrValueIfExists(cf, r, "id", e, "id");
+
+        cf.column("geom");
+
+        {
+            Element pointNode = getFirstNodeByTagName(e, "Point");
+            if (pointNode != null) {
+                Element coordsElt = getFirstNodeByTagName(pointNode, "coordinates");
+                // Mandatory
+                String coordsTxt = getTextContent(coordsElt);
+                String[] chunks = coordsTxt.split(",");
+
+                Coords coords = new Coords(Double.parseDouble(chunks[1]), Double.parseDouble(chunks[0]));
+                r.put(cf.column("geom"), coords.toWKT());
+            }
+        }
+
+        {
+            Element linestringNode = getFirstNodeByTagName(e, "LineString");
+            if (linestringNode != null) {
+                Element coordsElt = getFirstNodeByTagName(linestringNode, "coordinates");
+                // Mandatory
+                String coordsTxt = getTextContent(coordsElt);
+                logger.info("Parse linestring: " + coordsTxt);
+                String[] points = StringUtils.splitByWholeSeparator(coordsTxt,  " ");
+
+                List<String> pointsStr = new ArrayList<>();
+                for (String point : points) {
+                    if (StringUtils.isBlank(point)) continue;
+                    logger.info("POINT: --" + point + "--");
+                    String[] chunks = point.split(",");
+                    pointsStr.add(chunks[1] + " " + chunks[0]);
+                }
+                r.put(cf.column("geom"), "LINESTRING(" + StringUtils.join(pointsStr, ",") + ")");
```

Comment:
Here, test that pointsStr has more than 2 elements. As my grandpa said, "Can't create a line out of one point man!"